src/openms/source/FORMAT/ParamCWLFile.cpp

```diff
@@ -0,0 +1,155 @@
+// $Authors: Simon Gene Gottlieb $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/FORMAT/ParamCWLFile.h>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+// replaces the substrings inside a string
+// This method is 'static' so no external linkage occurs
+static std::string replaceAll(std::string str, const std::string& pattern, const std::string& replacement)
+{
+  size_t pos {0};
+  while ((pos = str.find(pattern, pos)) != std::string::npos)
+  {
+    str.replace(pos, pattern.size(), replacement);
+    pos += replacement.size();
+  }
+  return str;
+}
+
+namespace OpenMS
+{
+  void ParamCWLFile::load(const std::string& filename, Param& param)
+  {
+    // discover the name of the first nesting Level
+    // this is expected to result in something like "ToolName:1:"
+    auto traces = param.begin().getTrace();
+    std::string toolNamespace = traces.front().name + ":1:";
+
+    json jsonNode = json::parse(std::ifstream {filename});
+    if (!jsonNode.is_object())
+    {
+      OPENMS_LOG_WARN << "Ignoring JSON file '" << filename << "' because of unexpected data type. Expecting a dictionary as root type." << std::endl;
+      return;
+    }
+    for (auto child : jsonNode.items())
+    {
+      auto key = child.key();
+      key = toolNamespace + replaceAll(key, "__", ":"); // This converts __ to ':', but ':' would also be an accepted delimiter
+
+      auto node = child.value();
+      if (node.is_null())
+        continue; // No value given
+      if (!param.exists(key))
+      {
+        OPENMS_LOG_WARN << "Unknown (or deprecated) Parameter '" << key << "' given in outdated parameter file! Ignoring parameter." << std::endl;
+        continue;
+      }
+      auto const& entry = param.getEntry(key);
+      auto value = entry.value;
+      if (entry.value.valueType() == ParamValue::ValueType::STRING_VALUE)
+      {
+        if (entry.valid_strings.size() == 2 && entry.valid_strings[0] == "true" && entry.valid_strings[1] == "false")
+        {
+          value = node.get<bool>() ? "true" : "false";
+        }
+        else if (entry.tags.count("input file") || entry.tags.count("output file"))
+        {
+          value = node["path"].get<std::string>();
+        }
+        else
+        {
+          value = node.get<std::string>();
+        }
+      }
+      else if (entry.value.valueType() == ParamValue::ValueType::INT_VALUE)
+      {
+        value = node.get<int64_t>();
+      }
+      else if (entry.value.valueType() == ParamValue::ValueType::DOUBLE_VALUE)
+      {
+        value = node.get<double>();
+      }
+      else if (entry.value.valueType() == ParamValue::ValueType::STRING_LIST)
+      {
+        std::vector<std::string> list;
+        if (entry.tags.count("input file") || entry.tags.count("output file"))
+        {
+          for (auto child : node)
```

Comment:
this will trigger a CI error, since this may be a copy. use `const auto&` to avoid